src/btree/bt_split.c

```diff
@@ -2318,6 +2320,19 @@ __wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi)
     __wt_page_modify_clear(session, page);
     if (!F_ISSET(S2C(session)->cache, WT_CACHE_EVICT_SCRUB) || multi->supd_restore)
         F_SET_ATOMIC_16(page, WT_PAGE_EVICT_NO_PROGRESS);
+
+    /* If there's an address, copy it. */
+    if (multi->addr.block_cookie != NULL) {
+        WT_ERR(__wt_calloc_one(session, &addr));
+        WT_TIME_AGGREGATE_COPY(&addr->ta, &multi->addr.ta);
+        WT_ERR(__wt_memdup(
+          session, multi->addr.block_cookie, multi->addr.block_cookie_size, &addr->block_cookie));
+        addr->block_cookie_size = multi->addr.block_cookie_size;
+        addr->type = multi->addr.type;
+        __wt_ref_addr_free(session, ref);
+        ref->addr = addr;
+    }
```

Comment:
Higher-level design question: does giving the ref an address mean it's going to get written out later? Trying to piece the story together from the ticket + PR.

Reply:
Yes, checkpoint may write that address in the internal page. We also need to be able to free the address when a new reconciliation happen.